sqlglot/dialects/mysql.py

```diff
@@ -867,3 +867,29 @@ def chr_sql(self, expression: exp.Chr) -> str:
             charset = expression.args.get("charset")
             using = f" USING {self.sql(charset)}" if charset else ""
             return f"CHAR({this}{using})"
+
+        def timestamptrunc_sql(self, expression: exp.TimestampTrunc) -> str:
+            this = self.sql(expression, "this")
+            unit = expression.args.get("unit")
+            unix_timestart = "'1900-01-01 00:00:00'"
+
+            mysql_unit = unit
+            if isinstance(unit, exp.Var):
+                # Match abbrevations, most coming from Snowflake
```

Comment:
This shouldn't be needed, abbreviated units are [canonicalized](https://github.com/tobymao/sqlglot/blob/e82a30b6563547daea0bb087e1b6b5bf3b0532d3/sqlglot/dialects/snowflake.py#L144) at parse time for Snowflake. If for some reason they weren't in what you've tested, then we should fix that instead.